### DISCUSSION TITLE
Spatially inhomogeneous density mechanisms?

### DISCUSSION THREAD

**Discussion by espenhgn** [arbor-sim/arbor#1581] [Q&A] [answered]
Hi, 
I didn't see this pop up in the documentation or list of issues or elsewhere: Are there currently any ways of varying parameters of density mechanisms (`g` of `pas` etc.) dependent on location (from `place_pwlin`, relative location on branch or similar)? 

The decor examples shown here https://docs.arbor-sim.org/en/latest/concepts/decor.html?highlight=decor.paint#cablecell-properties for instance imply that one can only define constant densities per region. 

Specifically I would like to define a membrane current which parameters depends on some function `f(z)` where `z` is location on the z-axis. 

Edit: For some motivation: it is well known that densities of certain channels, for instance Ih may display an exponential like dependence on the distance to the soma in pyramidal cells: https://doi.org/10.1523/JNEUROSCI.3664-05.2006  

Thanks in advance.

**Comment by brenthuisman** [comment]
Using `arbor.mechanism` you can create your own function that varies any range parameter. The documentation is [here](https://docs.arbor-sim.org/en/latest/concepts/decor.html?highlight=axial#density-mechanisms). Or have you already seen this and is it not what you mean?

[edit] Ah, wait, I think this is not what you mean.

**Reply by espenhgn** [reply]
No, I didn't think that page described what I wanted :-)

**Reply by brenthuisman** [reply]
The short answer is: not possible ATM. It is on our TODO, but we discovered we don't have a clear issue for it. If you want, you can make one (you already wrote down clear motivation).

**Reply by espenhgn** [reply]
Fine, I can create an issue. Thanks.

**Comment by thorstenhater** [comment, marked as answer]
Hi @espenhgn,

we added this feature under the name 'Inhomogeneous Parameters' (https://docs.arbor-sim.org/en/stable/concepts/decor.html#scaled-mechanisms). Is that closer to what you had in mind?

**Reply by espenhgn** [reply]
Great, thanks a lot!
